pulsar-broker/src/main/java/org/apache/pulsar/broker/lookup/TopicLookupBase.java

```diff
@@ -127,6 +131,113 @@ protected void internalLookupTopicAsync(TopicName topicName, boolean authoritati
+                                }
+                                List<LookupData> lookupDataList = new ArrayList<>();
+                                for (CompletableFuture<Optional<LookupResult>> partitionFuture : futureList) {
+                                    partitionFuture.thenAccept(optionalResult -> {
+                                        if (optionalResult == null || !optionalResult.isPresent()) {
+                                            log.warn("No broker was found available for topic {}", topicName);
+                                            completeLookupResponseExceptionally(asyncResponse,
+                                                    new WebApplicationException(Response.Status.SERVICE_UNAVAILABLE));
+                                            return;
+                                        }
+
+                                        LookupResult result = optionalResult.get();
+                                        // We have found either a broker that owns the topic, or a broker to which we should redirect the client to
+                                        if (result.isRedirect()) {
+                                            boolean newAuthoritative = result.isAuthoritativeRedirect();
+                                            URI redirect;
+                                            try {
+                                                String redirectUrl = isRequestHttps() ? result.getLookupData().getHttpUrlTls()
+                                                        : result.getLookupData().getHttpUrl();
+                                                checkNotNull(redirectUrl, "Redirected cluster's service url is not configured");
+                                                String lookupPath = topicName.isV2() ? LOOKUP_PATH_V2 : LOOKUP_PATH_V1;
+                                                redirect = new URI(String.format("%s%s%s?authoritative=%s", redirectUrl, lookupPath,
+                                                        topicName.getLookupName(), newAuthoritative));
+                                            } catch (URISyntaxException | NullPointerException e) {
+                                                log.error("Error in preparing redirect url for {}: {}", topicName, e.getMessage(), e);
+                                                completeLookupResponseExceptionally(asyncResponse, e);
+                                                return;
+                                            }
+                                            if (log.isDebugEnabled()) {
+                                                log.debug("Redirect lookup for topic {} to {}", topicName, redirect);
+                                            }
+                                            completeLookupResponseExceptionally(asyncResponse,
+                                                    new WebApplicationException(Response.temporaryRedirect(redirect).build()));
```

Comment:
If redirect the partitioned topic lookup requests, the target broker will handle the partitioned topic lookup request right? So this may lead to redirect back again, the request can't complete.

Reply:
I'll check this later